tests/common/include/tests/common/tools/helix_gun.hpp

```diff
@@ -41,36 +62,109 @@ class helix_gun {
             // Get vz over vt in new coordinate
             _vz_over_vt = pz / getter::norm(pT);
         }
+    }
 
-        _scaler = _R * std::sqrt(1 + std::pow(_vz_over_vt, 2));
+    scalar radius() const { return _R; }
 
-        // get new y direction
-        scalar q = _vertex.qop() > 0 ? 1 : -1.;
-        vector3 y = -1 * q * vector::normalize(pT);
-        _ey = vector::normalize(y);
+    point3 operator()(scalar s) const { return this->pos(s); }
 
-        // get new x direction
-        _ex = vector::cross(_ey, _ez);
-    }
+    // position after propagating the path length of s
+    point3 pos(scalar s) const {
 
-    scalar radius() const { return _R; }
+        // Handle the case of pT ~ 0
+        if (_vz_over_vt == std::numeric_limits<scalar>::infinity()) {
+            return _vertex.pos() + s * _h0;
+        }
 
-    point3 operator()(scalar path_length) const {
+        point3 ret = _vertex.pos();
+
+        ret = ret + _delta / _K * (_K * s - std::sin(_K * s)) * _h0;
+
+        ret = ret + std::sin(_K * s) / _K * _t0;
+
+        ret = ret + _alpha / _K * (1 - std::cos(_K * s)) * _n0;
+
+        return ret;
+    }
+
+    // tangentional vector after propagating the path length of s
+    vector3 dir(scalar s) const {
 
         // Handle the case of pT ~ 0
         if (_vz_over_vt == std::numeric_limits<scalar>::infinity()) {
-            return _vertex.pos() + path_length * _ez;
+            return _vertex.dir();
         }
 
-        // Requested path length is coverted to parameterized value (t) by
-        // dividing it with scaling factor
-        scalar t = path_length / _scaler;
+        vector3 ret{0, 0, 0};
+
+        ret = ret + _delta * (1 - std::cos(_K * s)) * _h0;
+        ret = ret + std::cos(_K * s) * _t0;
+        ret = ret + _alpha * std::sin(_K * s) * _n0;
+
+        return ret;
+    }
+
+    // transport jacobian after propagating the path length of s
+    free_matrix jacobian(scalar s) const {
+
+        free_matrix ret =
+            matrix_operator().template zero<e_free_size, e_free_size>();
 
-        vector3 X = (_R * std::cos(t) - _R) * _ex;
-        vector3 Y = (_R * std::sin(t)) * _ey;
-        vector3 Z = (_R * _vz_over_vt * t) * _ez;
+        const matrix_type<3, 3> I33 =
+            matrix_operator().template identity<3, 3>();
+        const matrix_type<3, 3> Z33 = matrix_operator().template zero<3, 3>();
+        /// Get drdr
```

Comment:
Just a minor thing, but why are these doc comments and virtually all other comments aren't?

Reply:
It was just brain-less random commeting